src/Page/Submission/Team1/Algorithm.js

```diff
@@ -0,0 +1,185 @@
+// Sherman Grewal
+// Oliver Lingertat
+// Lucy Cheng
+// Navya Mehta
+import React, { Component } from 'react';
+import { render } from "react-dom";
+import StockStats from '../../../Data/stocks';
+
+class stockobj extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+    keys: Object.keys(StockStats), 
+    dates: StockStats.map(function(x) {
+        return x.map(function(day) {
+          return day.Date;
+        });
+      }),
+    closeVal: StockStats.map(function(x) {
+        return x.map(function(day) {
+          return day.Close;
+        });
+      }),
+    sma19: this.sma19(),
+    ema19: this.ema19(),
+    ema39: this.ema39(),
+    algval: this.mainalg()
+    };
+    this.InitFunction=this.InitFunction.bind(this);
+  }
+
+  InitFunction() {
+    this.props.childHandler(this.state);
+  }
+
+  render() {
+    return (
+        this.InitFunction
+      )
+  }
+
+  sma19() {
+    let closearr = StockStats.map(function(x) {
+        return x.map(function(day) {
+          return day.Close;
+        });
+      });
+    let sma = closearr.map(function(stock) {
+      let temparr = stock;
+      for (let i=0; i<19; i++)
+      {
+        temparr[i] = 0;
+      }
+      for (let i=19; i < 365; i++)
+      {
+        let temp = 0;
+        for (let j = i-19; j<i; j++)
+        {
+          temp = temp + stock[j];
+        }
+        temparr[i] = temp / 19;
+      }
+      return temparr;
+    });
+    return sma;
+  }
+  ema19() {
+    let expmul = 0.1;
+    let closearr = StockStats.map(function(x) {
+        return x.map(function(day) {
+          return day.Close;
+        });
+      });
+    let ema = closearr.map(function(stock) {
+      let tempema = stock;
+      let temp = 0;
+      for (let i=0; i<19; i++)
+      {
+        temp = temp + tempema[i];
+        tempema[i] = 0;
+      }
+      tempema[19] = temp / 19;
+      for (let i=20; i<365; i++)
+      {
+        tempema[i] = expmul * (tempema[i]-tempema[i-1]) + tempema[i-1];
+      }
+      return tempema;
+    });
+    return ema;
+  }
+  ema39() {
```

Comment:
I see a lot of duplicate code here (ema19 and ema39).
Suggestion: implement a ema and sma function which takes the day as a parameter. This implementation can simplify a lot of your code.